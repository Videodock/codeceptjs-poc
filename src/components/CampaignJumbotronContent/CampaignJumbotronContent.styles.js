import { createStyles } from '@material-ui/styles';

export default ({ palette, typography, spacing }) => createStyles({
  daysToGo: {
    color     : palette.primary.main,
    fontSize  : typography.pxToRem(26),
    fontWeight: 900,
  },
  deadline: {
    fontSize: typography.pxToRem(14),
  },
  buttons: {
    marginTop : spacing(),
    display   : 'flex',
    alignItems: 'center',
  },
  primaryButton: {
    color        : palette.common.white,
    padding      : '8px 32px',
    fontSize     : typography.pxToRem(16),
    letterSpacing: '0.66px',
  },
});
