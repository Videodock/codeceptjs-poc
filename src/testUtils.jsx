import React from 'react';
import { Provider } from 'react-redux';
import { render } from '@testing-library/react';
import { ThemeProvider } from '@material-ui/styles';
import { I18nextProvider } from 'react-i18next';
import { ApolloProvider } from '@apollo/client';
import { BrowserRouter as Router } from 'react-router-dom';
import 'url-search-params-polyfill';
import i18n from '../src/i18n';
import client from './graphql';
import theme from './theme';
import store from './redux';

const wrapper = ({ children }) => (
  <ApolloProvider client={client}>
    <Provider store={store}>
      <ThemeProvider theme={theme}>
        <I18nextProvider i18n={i18n}>
          <Router>
            {children}
          </Router>
        </I18nextProvider>
      </ThemeProvider>
    </Provider>
  </ApolloProvider>
);

const customRender = (ui, options) => render(ui, { wrapper, ...options });

export { customRender as render };
