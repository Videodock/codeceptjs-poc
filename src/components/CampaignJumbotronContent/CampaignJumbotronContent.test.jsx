import React from 'react';
import { render } from '../../testUtils';
import CampaignJumbotronContent from './CampaignJumbotronContent';

describe('<CampaignJumbotronContent />', () => {
  test('renders and matches snapshot', () => {
    const { container } = render(<CampaignJumbotronContent />);

    expect(container.firstChild).toMatchSnapshot();
  });

  // add your unit tests here
});
