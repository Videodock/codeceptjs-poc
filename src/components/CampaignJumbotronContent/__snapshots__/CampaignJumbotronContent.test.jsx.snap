// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CampaignJumbotronContent /> renders and matches snapshot 1`] = `
<div>
  <p
    class="MuiTypography-root makeStyles-daysToGo-1 MuiTypography-body1"
  >
    33 days to go
  </p>
  <p
    class="MuiTypography-root makeStyles-deadline-2 MuiTypography-body1"
  >
    Deadline: 16 May 2020
  </p>
  <div
    class="makeStyles-buttons-3"
  >
    <a
      aria-disabled="false"
      class="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-primaryButton-4 MuiButton-containedPrimary"
      href="/campaign/e63fc3c5-f84e-4a64-9d5b-98a49dd4680c"
      role="button"
      tabindex="0"
    >
      <span
        class="MuiButton-label"
      >
        Join campaign
      </span>
      <span
        class="MuiTouchRipple-root"
      />
    </a>
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-text"
      tabindex="0"
      type="button"
    >
      <span
        class="MuiButton-label"
      >
        <span
          class="MuiButton-startIcon MuiButton-iconSizeMedium"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root"
            focusable="false"
            role="presentation"
            viewBox="0 0 24 24"
          >
            <path
              d="M12 22c1.1 0 2-.9 2-2h-4c0 1.1.89 2 2 2zm6-6v-5c0-3.07-1.64-5.64-4.5-6.32V4c0-.83-.67-1.5-1.5-1.5s-1.5.67-1.5 1.5v.68C7.63 5.36 6 7.92 6 11v5l-2 2v1h16v-1l-2-2z"
            />
          </svg>
        </span>
        Subscribe for updates
      </span>
      <span
        class="MuiTouchRipple-root"
      />
    </button>
  </div>
</div>
`;
